patches/libarchive/archive_write_add_filter_gzip.c.patch

```diff
@@ -0,0 +1,58 @@
+--- a/libarchive/archive_write_add_filter_gzip.c
++++ b/libarchive/archive_write_add_filter_gzip.c
+@@ -58,6 +58,7 @@ archive_write_set_compression_gzip(struct archive *a)
+ struct private_data {
+ 	int		 compression_level;
+ 	int		 timestamp;
++	unsigned char 	os;
+ #ifdef HAVE_ZLIB_H
+ 	z_stream	 stream;
+ 	int64_t		 total_in;
+@@ -106,6 +107,7 @@ archive_write_add_filter_gzip(struct archive *_a)
+ 		archive_set_error(&a->archive, ENOMEM, "Out of memory");
+ 		return (ARCHIVE_FATAL);
+ 	}
++	data->os = 3; /* default Unix */
+ 	f->data = data;
+ 	f->open = &archive_compressor_gzip_open;
+ 	f->options = &archive_compressor_gzip_options;
+@@ -166,6 +168,30 @@ archive_compressor_gzip_options(struct archive_write_filter *f, const char *key,
+ 		return (ARCHIVE_OK);
+ 	}
+ 
++	if (strcmp(key, "os") == 0) {
```

Comment:
whats the story here?

Reply:
gzip has a byte reserved for specifying which operating system it was compressed on. libarchive hardcodes it to 3 (Unix), and the package managers set it to 255 (unknown os).